src/components/SignIn/useSignIn.ts

```diff
@@ -0,0 +1,38 @@
+import axios from "../../utils/axios";
+import { useState } from "react";
+import { useRouter } from "next/router";
+import { useToast } from "@chakra-ui/react";
+
+export default function useSignIn() {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const toast = useToast();
+  const router = useRouter();
+  const signInRequest = async (username: string, password: string) => {
+    const response = await axios.post("/staff/signin", {
+      username,
+      password,
+    });
+    return response;
+  };
+  const handleSignInSubmit = async (data: { username: string; password: string }) => {
+    setIsLoading(true);
+    try {
+      const response = await signInRequest(data.username, data.password);
```

Comment:
You have deconstructed `username` and `password` from `data`, why not use them directly

Reply:
done